### PR TITLE
Compile Mono for other platforms

### PR DESCRIPTION
**Pull request by joshpeterson**
This pull request allows our Mono 5.0 branch to compile for some of the additional platforms Unity supports which we need for IL2CPP on Mono. Specifically, we're compiling eglib, libmonoutils, and libmonoruntime with the Unity PAL.

With these changes, the Mono build succeeds for all of the current platforms as well.

### CODE REVIEW

File: mono/utils/mono-dl.c [hunk 1]
```diff
@@ -37,11 +37,11 @@ struct MonoDlFallbackHandler {
 static GSList *fallback_handlers;
 
 
-/*
```
Comment (joshpeterson) [inline mono/utils/mono-dl.c]: It seems our formatting tool removed trailing whitespace on a number of lines in this file. I think this is generally a good thing, but I did not change this line or any of them below in any meaningful way. Should I put back the trailing whitespace so as not make make the diff more confusing?

File: mono/utils/mono-dl.c [hunk 2]
```diff
@@ -23,7 +23,7 @@
 #include <LibraryLoader-c-api.h>
 #endif
 
-#ifndef HOST_WIN32 && defined (HAVE_DL_LOADER)
+#if !defined (HOST_WIN32) && defined (HAVE_DL_LOADER)
```
Comment (joshpeterson) [inline mono/utils/mono-dl.c]: This is the only real change to this file.

File: mono/metadata/mono-security.c [hunk 3]
```diff
@@ -76,6 +76,8 @@ GetTokenName (uid_t uid)
 {
 	gchar *uname = NULL;
 
+#ifdef HAVE_PWD_H
```
Comment (joncham) [inline mono/metadata/mono-security.c]: Are changes like this worth upstreaming?

File: mono/utils/mono-membar.h [hunk 4]
```diff
@@ -60,6 +60,23 @@ static inline void mono_memory_write_barrier (void)
 {
 	mono_memory_barrier ();
 }
+#elif  defined(__x86_64__)
+#ifndef _MSC_VER
```
Comment (joncham) [inline mono/utils/mono-membar.h]: Is there a positive check for this? `__GNUC__` or something?

File: mono/utils/mono-proclib.c [hunk 5]
```diff
@@ -29,8 +29,10 @@
 #ifdef HAVE_SYS_SYSCTL_H
 #include <sys/sysctl.h>
 #endif
+#if !defined(PLATFORM_UNITY)
```
Comment (joncham) [inline mono/utils/mono-proclib.c]: No `#ifdef HAVE_SYS_RESOURCE_H`?

File: mono/metadata/mono-security.c (hunk 3 above)
Comment (joshpeterson) [inline mono/metadata/mono-security.c]: Yes, I've already upstreamed these changes, actually.

File: mono/utils/mono-proclib.c (hunk 5 above)
Comment (joshpeterson) [inline mono/utils/mono-proclib.c]: Good point, I've changed it to use the check you suggested instead.

File: mono/utils/mono-membar.h (hunk 4 above)
Comment (joshpeterson) [inline mono/utils/mono-membar.h]: I've changed this to be a positive check.

**Pull request merged**